### DISCUSSION TITLE
Second instance with next month view?

### DISCUSSION THREAD

**Discussion by spxhub** [MMRIZE/MMM-CalendarExt3#120] [Q&A]
Hi,

I have the following questions:

1.) If I wanted to achieve the following: With a first instance display the current month in month-mode and with a second instance display the following month also in month-mode. How would I do this (if possible)? Also, would it be possible to start the following month with the first full week so that there is no week (or days) shown twice (once at the end of the current month and once at the beginning of the following month)

2.) My current month-mode view shows 5 cw's. Is there an option to extend month-mode to show say 6 weeks? Like also having the first full week of the following month shown in the month-view of the current month.

Thanks for your comments!
And thanks for this great module!

**Comment by eouia** [comment]
1. Multi instances are possible. Just add the module again in config.js. It would be better to assign different `instanceId` to distinguish each other. See the README about that config value.
2. You can dynamically change the month step on the fly, but it is not possible at the start. I'll consider that feature on next update. (Nobody has requested that feature yet.)
3. I guess you may want a `sequential 2-month view vertically arranged`. Sorry, that kind of view is not prepared. But I doubt it's worth it. 
4. Instead, you can use an alternative `week` or `day` mode view to adjust the number of weeks displayed.

**Reply by spxhub** [reply]
Thank you for your reply!
So if I understand correctly answer #2 means that currently I can not start an instance with the following month? I could, however, change the displayed month of a instance on the fly (with a notification i suppose), right?
Although, I could implement it that way (via rotary encoder or key press) it would be great to have an option to display the following month in a second instance permanently from the start.

**Reply by eouia** [reply]
Yep I’ll consider to implement that. To be honest, I thought It had that feature already, but I have missed the real coding. :)

**Reply by eouia** [reply]
I updated the new version a few days ago. try it.

**Reply by spxhub** [reply]
Thank you very much! I installed it already and it is working flawlessly so far.

**Reply by cmtao** [reply]
> 1. Multi instances are possible. Just add the module again in config.js. It would be better to assign different `instanceId` to distinguish each other. See the README about that config value.
> 2. You can dynamically change the month step on the fly, but it is not possible at the start. I'll consider that feature on next update. (Nobody has requested that feature yet.)
> 3. I guess you may want a `sequential 2-month view vertically arranged`. Sorry, that kind of view is not prepared. But I doubt it's worth it.
> 4. Instead, you can use an alternative `week` or `day` mode view to adjust the number of weeks displayed.

If possible, could you add buttons to allow the monthly calendar view to switch next month(s) and previous month(s)?

**Reply by eouia** [reply]
@cmtao 
> If possible, could you add buttons to allow the monthly calendar view to switch next month(s) and previous month(s)?

That is the out of this module's work. There are already many modules to manipulate other module. You can use one of those modules for your purpose to collaborate with this module.
If you have still trouble, submit your issue as a new post, I missed this comment for a while.
